blob/fileblob/fileblob.go

```diff
@@ -57,35 +59,18 @@ func OpenBucket(dir string) (*blob.Bucket, error) {
 	return blob.NewBucket(&bucket{dir}), nil
 }
 
-// resolvePath converts a key into a relative filesystem path. It guarantees
-// that there will only be one valid key for a given path and that the resulting
-// path will not reach outside the directory.
-func resolvePath(key string) (string, error) {
-	for _, c := range key {
-		if !('A' <= c && c <= 'Z' || 'a' <= c && c <= 'z' || '0' <= c && c <= '9' || c == '/' || c == '.' || c == ' ' || c == '_' || c == '-') {
-			return "", fmt.Errorf("contains invalid character %q", c)
-		}
-	}
-	if cleaned := slashpath.Clean(key); key != cleaned {
-		return "", fmt.Errorf("not a clean slash-separated path")
-	}
-	if slashpath.IsAbs(key) {
-		return "", fmt.Errorf("starts with a slash")
-	}
-	if key == "." {
-		return "", fmt.Errorf("invalid path \".\"")
-	}
-	if strings.HasPrefix(key, "../") {
-		return "", fmt.Errorf("starts with \"../\"")
-	}
-	return filepath.FromSlash(key), nil
+func escape(s string) string {
+	s = url.PathEscape(s)
+	// Unescape "/".
+	return strings.Replace(s, "%2F", "/", -1)
```

Comment:
This should be replaced with the OS's file separator, which isn't always slash.

Reply:
Done. It will be nice to get Windows tests for this package.